libsel4/sel4_arch_include/aarch64/sel4/sel4_arch/types.h

```diff
@@ -16,14 +18,17 @@ typedef seL4_CPtr seL4_ARM_VSpace;
 
 typedef struct seL4_UserContext_ {
     /* frame registers */
-    seL4_Word pc, sp, spsr, x0, x1, x2, x3, x4, x5, x6, x7, x8, x16, x17, x18, x29, x30;
-    /* other integer registers */
-    seL4_Word x9, x10, x11, x12, x13, x14, x15, x19, x20, x21, x22, x23, x24, x25, x26, x27, x28;
+    seL4_Word spsr; /* Status register - 64-bit integer */
+    /* Registers that could hold pointers (and CHERI capabilities, when enabled) */
+    seL4_Ptr pc, sp, x0, x1, x2, x3, x4, x5, x6, x7, x8, x16, x17, x18, x29, x30;
```

Comment:
Dong it this way guarantees that `pc` is not properly aligned. Probably move `spsr` down.

Reply:
The toolchain will always align capabilities to their sizes, so it's not an issue here. I left the same ordering as sel4debug has some asserts for those register offsets, but could just change that as well if needed.

Reply:
`ldp`, `stp` often take one cycle longer per instruction if the two registers are not aligned on `2 * REGSIZE`.  And we don't want toolchain implicit padding in general in seL4 code anyway.